css/css-contain/container-queries/animation-container-size.html

```diff
@@ -31,13 +31,15 @@
 <script>
   setup(() => assert_implements_container_queries());
 
-  test(() => {
+  promise_test(async () => {
+    if (waitForPolyfill) await waitForPolyfill();
```

Comment:
I'm not too familiar with how async functions work, but could this be wrapped in a separate function like this to avoid if-statements all over the place?:
```
async waitForPolyfillIfNeeded() {
  if (waitForPolyfill)
    await waitForPolyfill();
}
```
or will that have the side-effects Emilio talked about?

Reply:
Yeah, the only way to avoid the side effects that Emilio mentioned when using `async` is to conditionally `await`, as promise callbacks are always asynchronous.

Another option would be to turn this into a generator function instead:

```js
cq_polyfillable_test(function*() {
  yield; // Equivalent to waiting if needed, but no condition required
  // do more synchronous work
});
```

That would let us move the asynchronous work into a `cq_polyfillable_test` function, but this might introduce more confusion than it's worth. Let me know what you think.

Reply:
FWIW, I think the conditional version is easier to follow (e.g. `if (waitForPolyfill)`)